challenges/making_anagrams.py

```diff
@@ -0,0 +1,12 @@
+from functools import reduce
+
+def number_needed(a, b):
+    """ Returns the number of characters you must delete to make the two strings anagrams of each other """
+    letters = {}
+    for l in a:
+        letters[l] = letters[l] + 1 if l in letters and letters[l] else 1
+
+    for l in b:
+        letters[l] = letters[l] - 1 if l in letters and letters[l] else -1
+
+    return reduce(lambda x, y: abs(x) + abs(y), letters.values(), 0)
```

Comment:
Agreed, while technically slower, 1) it is more python-y and 2) who knows what the python compiler might do behind the scenes to optimize this